lib/experimental/blocks.php

```diff
@@ -121,3 +121,69 @@ function gutenberg_register_metadata_attribute( $args ) {
 	return $args;
 }
 add_filter( 'register_block_type_args', 'gutenberg_register_metadata_attribute' );
+
+
+$gutenberg_experiments = get_option( 'gutenberg-experiments' );
+if ( $gutenberg_experiments && array_key_exists( 'gutenberg-connections', $gutenberg_experiments ) ) {
+	/**
+	 * Renders the block meta attributes.
+	 *
+	 * @param string   $block_content Block Content.
+	 * @param array    $block Block attributes.
+	 * @param WP_Block $block_instance The block instance.
+	 */
+	function gutenberg_render_custom_sources( $block_content, $block, $block_instance ) {
+		$meta_custom_source = require __DIR__ . '/custom-sources/meta.php';
+		$block_type         = $block_instance->block_type;
+
+		// Whitelist of the block types that support custom sources
+		// Currently, we only allow the Paragraph and Image blocks to use custom sources.
+		if ( ! in_array( $block['blockName'], array( 'core/paragraph', 'core/image' ), true ) ) {
+			return $block_content;
+		}
```

Comment:
This serves as a "whitelist" of the block attributes:

https://github.com/WordPress/gutenberg/blob/f0bcb0fb04f9412fb67303f12c101c073a7bc0c6/packages/block-editor/src/hooks/custom-fields.js#L67-L68

Do you think that's sufficient?

Reply:
I suppose we could only check the `content` attribute of the Paragraph and the `url` of the Image, instead of iterating over the attributes on this line: 

https://github.com/WordPress/gutenberg/blob/60fa0613981cbd1cc45010cefdae40d4a49be509/lib/experimental/blocks.php#L161

Reply:
> I don't know what the best mechanism is, and I guess we will have to iterate it, but right now if I add a connection manually to the block (without using the Block Editor), I could include any block attribute and the PHP logic will try to replace that, right?

Yep, that's true. OK, makes sense that we create an explicit, more centralized whitelist in the `blocks.php` file.

Reply:
done in 49d5cb6e3c6a167825af85cf00e40370cf767988